### ISSUE TITLE
Game over but the game can still be played

### ISSUE THREAD

**Issue by hteumeuleu** [hteumeuleu/2048#1]
From user Bugpy on itch: 

> I'm getting a bug where the game over screen appears but I can continue the game behind. Sometimes it ends the game with possible moves left and although it's hard to see, I can still continue whilst the "Game Over" dialog is up and my score still goes up.

[![Screenshot 1](https://i.imgur.com/ygNPiYBm.jpeg)](https://i.imgur.com/ygNPiYB.jpeg) [![Screenshot 1](https://i.imgur.com/r0mmHWXm.jpeg)](https://i.imgur.com/r0mmHWX.jpeg) [![Screenshot 1](https://i.imgur.com/TRCQ7GGm.jpeg)](https://i.imgur.com/TRCQ7GG.jpeg) [![Screenshot 1](https://i.imgur.com/ULwTsqHm.jpeg)](https://i.imgur.com/ULwTsqH.jpeg)

**Comment by hteumeuleu** [comment]
Here’s a `save.json` data to reproduce the problem after a few moves:

```json
[
	"4,2,4,2,4,4,256,8,x,8,128,2,8,64,2,4",
	240
]
```

**Comment by hteumeuleu** [comment]
Here’s another `save.json` to reproduce the problem with just one move: 

```json
[
	"x,2,8,16,4,256,16,2,32,64,128,4,4,2,4,2",
	420
]
```

**Comment by hteumeuleu** [comment]
Should be fixed in v1.0.3. 

The problem was due to how the game was initialised from a previous save. Tile’s values would be set as a `String` instead of a `Number`. Then in the end of the game, when the grid is full, the game would incorrectly compare `String` values (from tiles restored from a previous save) with `Number` values (from tiles created during the game after loading).
